plugins/media/wowza/services/LiveConversionProfileService.php

```diff
@@ -174,16 +194,101 @@ public function serveAction($streamName, $hostname = null)
 		
 		return new kRendererString($dom->saveXML(), 'text/xml');
 	}
-
-	protected function appendLiveParams(LiveStreamEntry $entry, WowzaMediaServerNode $mediaServer = null, SimpleXMLElement $encodes, liveParams $liveParams)
+	
+	private function calculateFlavorHeight($flavorResolution, $ingestParameters)
+	{
+		return $flavorResolution['width'] * $ingestParameters['height'] / $ingestParameters['width'];
+	}
+	
+	private function isFlavorCompatibile($ingestParameters, $flavorBitrate, $flavorResolution)
+	{
+		$flavorHeight = 0;
+		switch ($flavorResolution['fitMode'])
+		{
+			case 'match-source':
+				break;
+			case 'fit-height':
+				$flavorHeight = $flavorResolution['height'];
+				break;
+			case 'fit-width':
+				$flavorHeight = $this->calculateFlavorHeight($flavorResolution, $ingestParameters);
+				break;
+		}
+		
+		if (isset($ingestParameters['height']) && $ingestParameters['height'] < $flavorHeight)
+		{
+			return false;
+		}
+		else if (isset($ingestParameters['videodatarate']) && $ingestParameters['videodatarate'] < $flavorBitrate)
+		{
+			return false;
+		}
+		
+		return true;
+	}
+	
+	private function getResolutionParameters($liveParams)
+	{
+		$resolutionObject = array(
+			'fitMode' => ''
+		);
+		if(!$liveParams->getWidth() && !$liveParams->getHeight())
+		{
+			$resolutionObject['fitMode'] = 'match-source';
+		}
+		elseif($liveParams->getWidth() && $liveParams->getHeight())
+		{
+			$resolutionObject['fitMode'] = 'fit-height';
+			$resolutionObject['width'] = $liveParams->getWidth();
+			$resolutionObject['height'] = $liveParams->getHeight();
+		}
+		elseif($liveParams->getWidth())
+		{
+			$resolutionObject['fitMode'] = 'fit-width';
+			$resolutionObject['width'] = $liveParams->getWidth();
+		}
+		elseif($liveParams->getHeight())
+		{
+			$resolutionObject['fitMode'] = 'fit-height';
+			$resolutionObject['height'] = $liveParams->getHeight();
+		}
+		
+		return $resolutionObject;
+	}
+	
+	private function checkMaxFramerate($ingestFramerate, $flavorMaxFramerate)
+	{
+		if ($flavorMaxFramerate)
+		{
+			$skipCount = ($ingestFramerate / $flavorMaxFramerate) - 1;
```

Comment:
why not just return ceil(($ingestFramerate / $flavorMaxFramerate) - 1   -  0.05)